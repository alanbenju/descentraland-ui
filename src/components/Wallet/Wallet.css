.Wallet .ui.card {
    padding: 16px;
  }
  
  .Wallet .error {
    margin-top: 20px;
    color: var(--primary);
  }
  