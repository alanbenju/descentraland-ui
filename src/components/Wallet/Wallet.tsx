import React from 'react'
import {
    Button,
    Card,
    Center,
    Footer,
    Header,
    Page,
} from 'decentraland-ui'
import { Props } from './Wallet.types'
import './Wallet.css'
import { Link, useLocation } from 'react-router-dom'

const Wallet: React.FC<Props> = ({
    address,
    balance,
    isConnected,
    onConnect,
    isConnecting,
    error,
}) => {

    let location = useLocation();

    console.log("wallet location", location)
    return (
        <>
            <Page className="Wallet">
                <Center>
                    {!isConnected ? (
                        <>
                            <Button primary onClick={onConnect} loading={isConnecting}>
                                Connect
                            </Button>
                            {error ? <p className="error">{error}</p> : null}
                        </>
                    ) : (
                        <Card>
                            <Header>Wallet</Header>
                            <p>
                                <strong>Address:</strong>&nbsp;
                                {address.slice(0, 6) + '...' + address.slice(-4)}
                            </p>
                            <p>
                                <strong>Balance:</strong>&nbsp;
                                {balance} DUMMY <Link to="/transfer" state={{ background: location }}>Transfer</Link>
                            </p>
                        </Card>
                    )}
                </Center>
            </Page>
            <Footer />
        </>
    )
}

export default Wallet
