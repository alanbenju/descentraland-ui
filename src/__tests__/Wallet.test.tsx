import { mount, configure } from 'enzyme';
import Wallet from '../components/Wallet/Wallet';
import { Link, MemoryRouter as Router } from 'react-router-dom';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
configure({adapter: new Adapter()});

const mockUseLocationValue = {
    pathname: "/",
    search: '',
    hash: '',
    state: null
}
jest.mock('react-router-dom', () => ({
    ...jest.requireActual("react-router-dom") as {},
    useLocation: jest.fn().mockImplementation(() => {
        return mockUseLocationValue;
    })
}));

describe('<Wallet />', () => {
    describe('Not connected wallet', () => {
        let component: any;
        let props:any;
        beforeAll(() => {
            props = {
                address: "null",
                balance: 0,
                isConnected: false,
                onConnect: jest.fn(),
                isConnecting: false,
                error: undefined
            }
            component = mount(<Wallet {...props}/>);
        })

        it('Render wallet not connected', () => {
            expect(component.html()).toMatchSnapshot();
        });
        it('Click on connect button', () => {
            let clickEvent = () => {
                component.find('button').simulate('click');
            };
            expect(clickEvent).not.toThrow();
            expect(props.onConnect).toBeCalled()
        });
       
    });
    describe('Connected wallet', () => {
        let component: any;
        let props:any;
        beforeAll(() => {
            props = {
                address: "0x123",
                balance: 100,
                isConnected: true,
                onConnect: jest.fn(),
                isConnecting: false,
                error: "null"
            }
            component = mount(<Router><Wallet {...props}/></Router>);
        })

        it('Render wallet connected', () => {
            expect(component.html()).toMatchSnapshot();
        });
        it('Link to transfer', () => {
            expect(component.find(Link).at(0).props().to).toBe('/transfer');
        });
       
    });
})