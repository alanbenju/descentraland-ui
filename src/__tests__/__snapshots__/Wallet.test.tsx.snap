// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wallet /> Connected wallet Render wallet connected 1`] = `"<div class=\\"dcl page Wallet\\"><div class=\\"ui container\\"><div class=\\"dcl center\\"><div class=\\"ui card\\"><div class=\\"ui header\\">Wallet</div><p><strong>Address:</strong>&nbsp;0x123...x123</p><p><strong>Balance:</strong>&nbsp;100 DUMMY <a href=\\"/transfer\\">Transfer</a></p></div></div></div></div><div class=\\"ui container dcl footer\\"><div class=\\"main-footer\\"><div role=\\"listbox\\" aria-expanded=\\"false\\" class=\\"ui upward dropdown dcl language-dropdown\\" tabindex=\\"0\\"><div aria-atomic=\\"true\\" aria-live=\\"polite\\" role=\\"alert\\" class=\\"divider text\\"><div class=\\"dcl language-icon-wrapper\\"><i class=\\"dcl language-icon en\\"></i><div class=\\"language-icon-label\\">English</div></div></div><i aria-hidden=\\"true\\" class=\\"dropdown icon\\"></i><div class=\\"right menu transition\\"><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"true\\" aria-selected=\\"true\\" class=\\"active selected item\\"><div class=\\"dcl language-icon-wrapper\\"><i class=\\"dcl language-icon en\\"></i><div class=\\"language-icon-label\\">English</div></div></div></div></div><div class=\\"links\\"><a href=\\"https://decentraland.org\\">Home</a><a href=\\"https://decentraland.org/privacy\\">Privacy Policy</a><a href=\\"https://decentraland.org/terms\\">Terms of Use</a><a href=\\"https://decentraland.org/content\\">Content Policy</a><a href=\\"https://decentraland.org/ethics\\">Code of Ethics</a></div></div><div class=\\"secondary-footer\\"><div class=\\"social-links\\"><a href=\\"https://dcl.gg/discord\\"><i class=\\"social-icon discord\\"></i></a><a href=\\"https://reddit.com/r/decentraland\\"><i class=\\"social-icon reddit\\"></i></a><a href=\\"https://github.com/decentraland\\"><i class=\\"social-icon github\\"></i></a><a href=\\"https://twitter.com/decentraland\\"><i class=\\"social-icon twitter\\"></i></a></div><div class=\\"copyright\\">© 2022 Decentraland</div></div></div>"`;

exports[`<Wallet /> Not connected wallet Render wallet not connected 1`] = `"<div class=\\"dcl page Wallet\\"><div class=\\"ui container\\"><div class=\\"dcl center\\"><button class=\\"ui primary button\\">Connect</button></div></div></div><div class=\\"ui container dcl footer\\"><div class=\\"main-footer\\"><div role=\\"listbox\\" aria-expanded=\\"false\\" class=\\"ui upward dropdown dcl language-dropdown\\" tabindex=\\"0\\"><div aria-atomic=\\"true\\" aria-live=\\"polite\\" role=\\"alert\\" class=\\"divider text\\"><div class=\\"dcl language-icon-wrapper\\"><i class=\\"dcl language-icon en\\"></i><div class=\\"language-icon-label\\">English</div></div></div><i aria-hidden=\\"true\\" class=\\"dropdown icon\\"></i><div class=\\"right menu transition\\"><div style=\\"pointer-events: all;\\" role=\\"option\\" aria-checked=\\"true\\" aria-selected=\\"true\\" class=\\"active selected item\\"><div class=\\"dcl language-icon-wrapper\\"><i class=\\"dcl language-icon en\\"></i><div class=\\"language-icon-label\\">English</div></div></div></div></div><div class=\\"links\\"><a href=\\"https://decentraland.org\\">Home</a><a href=\\"https://decentraland.org/privacy\\">Privacy Policy</a><a href=\\"https://decentraland.org/terms\\">Terms of Use</a><a href=\\"https://decentraland.org/content\\">Content Policy</a><a href=\\"https://decentraland.org/ethics\\">Code of Ethics</a></div></div><div class=\\"secondary-footer\\"><div class=\\"social-links\\"><a href=\\"https://dcl.gg/discord\\"><i class=\\"social-icon discord\\"></i></a><a href=\\"https://reddit.com/r/decentraland\\"><i class=\\"social-icon reddit\\"></i></a><a href=\\"https://github.com/decentraland\\"><i class=\\"social-icon github\\"></i></a><a href=\\"https://twitter.com/decentraland\\"><i class=\\"social-icon twitter\\"></i></a></div><div class=\\"copyright\\">© 2022 Decentraland</div></div></div>"`;
